src/devices/src/virtio/block/persist.rs

```diff
@@ -24,12 +24,34 @@ use crate::virtio::{DeviceState, TYPE_BLOCK};
 pub struct BlockState {
     id: String,
     partuuid: Option<String>,
+    #[version(
+        start = 3,
+        ser_fn = "block_ignore_flush_ser",
+        default_fn = "default_block_ignore_flush"
+    )]
+    ignore_flush: bool,
     root_device: bool,
     disk_path: String,
     virtio_state: VirtioDeviceState,
     rate_limiter_state: RateLimiterState,
 }
 
+impl BlockState {
+    fn block_ignore_flush_ser(&mut self, target_version: u16) -> VersionizeResult<()> {
+        if target_version < 3 && !self.ignore_flush {
+            return Err(VersionizeError::Semantic(
+                "Target version does not implement writethrough block cache.".to_owned(),
+            ));
```

Comment:
Should we make this backward-compatible, where the cache/flush just reverts to old behavior (also print a `warn!` or smth)?

Reply:
I am leaning towards no because of the risk that users might shoot themselves in the foot, but I can see how we can enable this and print a warning or something and let them handle it.

I guess we can do either, I'm on the fence about this. Other opinions?

Reply:
@firecracker-microvm/compute-capsule

Reply:
I went ahead and changed it so that it's only a warning now. Also added an integ test for this case.

@firecracker-microvm/compute-capsule please raise concerns if you have any.